src/subset.rs

```diff
@@ -220,7 +228,25 @@ impl<N: NodeIdT + Rand> Subset<N> {
                 return Ok(step);
             }
         };
-        self.broadcast_results.insert(proposer_id.clone(), value);
+        {
+            let ba_results = &self.ba_results;
```

Comment:
There's probably no need for the curly braces and for borrowing `self.ba_results` anymore.